lib/iron_bank/describe/excluded_fields.rb

```diff
@@ -41,33 +41,37 @@ def self.call(object_name:)
       def call
         remove_last_failure_fields until valid_query?
 
-        excluded_fields
+        excluded_fields - separate_query_fields
       end
 
       private
 
-      attr_reader :object_name, :excluded_fields, :last_failed_fields
+      attr_reader :object_name, :last_failed_fields
 
       def_delegators "IronBank.logger", :info
+      def_delegators :object, :separate_query_fields
 
       def initialize(object_name)
         @object_name        = object_name
-        @excluded_fields    = []
         @last_failed_fields = nil
       end
 
       def object
         IronBank::Resources.const_get(object_name)
       end
 
+      def excluded_fields
+        @excluded_fields ||= object.excluded_fields.dup
+      end
+
       def remove_last_failure_fields
         query_fields = object.query_fields
 
         failed_fields = query_fields.select do |field|
           last_failed_fields.any? { |failed| field.casecmp?(failed) }
         end
 
-        @excluded_fields += failed_fields
+        excluded_fields.push(*failed_fields)
```

Comment:
Could duplicates cause any issues here?

Reply:
I could call `#uniq` here, but the code should not push duplicate fields to the `excluded_fields` array. If we do, it's an error in our logic. I'd rather see duplicated fields in the output, which would tell me there might be an error here, instead of calling `uniq` and possibly not catching that issue.

Note: duplicate fields in the file are not an issue at runtime.